content/x-ray/x-ray.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "X-Ray Console"
+date: 2018-11-177T08:30:11-07:00
+weight: 13
+draft: false
+---
+
+We now have the example microservices deployed, so we are going to investigate our [Service Graph](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-servicegraph) and [Traces](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-traces) in [X-Ray section of the AWS Management Console](https://us-west-2.console.aws.amazon.com/xray/home?region=us-west-2#/service-map).
+
+The [Service map in the console](https://us-west-2.console.aws.amazon.com/xray/home?region=us-west-2#/service-map) provides a visual representation of the steps identified by X-Ray for a particular trace. Each resource that sends data to X-Ray within the same context appears as a service in the graph.
+
+![GitHub Edit](/images/x-ray/service_map.png)
+
+Next, go to the [traces section in the AWS Management Console](https://us-west-2.console.aws.amazon.com/xray/home?region=us-west-2#/traces) to view the execution times for the segments in the requests. At the top of the page, we can see the URL for the ELB endpoint and the corresponding traces below.
+
+![GitHub Edit](/images/x-ray/traces.png)
+
+If you click on the link on the left in the `Trace list` section you will see the overall execution time for the request, as well as a breakdown of the individual [segments](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-segments) in the request. In this visualization, you can see the front-end and back-end segments and a [subsegment](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-subsegments) named x-ray-sample-back-k8s-gen. In the [back-end service source code](https://github.com/rnzsgh/eks-workshop-x-ray-sample-back/blob/master/main.go#L33), we instrumented a subsegment that surrounds a random number generator.
+
+In the Go example, [the main segment](https://github.com/rnzsgh/eks-workshop-x-ray-sample-back/blob/master/main.go#L26) is initialized in the xray.Handler helper, which in turn sets all necessary information in the http.Request context struct, so that it can be used when initializing the subsegment.
+
```

Comment:
I would also add similar data here. For ex, in the below diagram we can see that frontend took average of 0.4ms to responded to requests ...